proto/osmosis/poolmanager/v2/query.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+package osmosis.poolmanager.v2;
+
+import "gogoproto/gogo.proto";
+
+import "google/api/annotations.proto";
+import "google/protobuf/any.proto";
+import "cosmos_proto/cosmos.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v19/x/poolmanager/client/queryprotov2";
+
+service Query {
+  // SpotPriceV2 defines a gRPC query handler that returns the spot price given
+  // a base denomination and a quote denomination.
+  // The returned spot price has 36 decimal places. However, some of
+  // modules perform sig fig rounding so most of the rightmost decimals can be
+  // zeroes.
+  rpc SpotPriceV2(SpotPriceRequest) returns (SpotPriceResponse) {
+    option (google.api.http).get =
+        "/osmosis/poolmanager/v2/pools/{pool_id}/prices";
+  }
+}
+
+// SpotPriceRequest defines the gRPC request structure for a SpotPrice
+// query.
+message SpotPriceRequest {
```

Comment:
shouldn't this also have a `V2` suffix? to avoid confusing it to v1's version: https://github.com/osmosis-labs/osmosis/blob/da6a1c59397177447bd25a21c6a164a428b21e4d/proto/osmosis/poolmanager/v1beta1/query.proto#L216.

same goes for `SpotPriceResponse`

Reply:
on the second thought, I guess it is fine, since we call it by specifying a package name, which has `v2` in it. still going to leave it here just in case